### PR TITLE
Efficient way to insert a number into a sorted array of numbers in Java

### PR DESCRIPTION
**Pull request by mee-omi**

### CODE REVIEW

File: core-java-modules/core-java-arrays-sorting/src/main/java/com/baeldung/sorted/operations/InsertElementInSortedArray.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        int[] newArr = new int[arr.length + 1];
+        
+        System.arraycopy(arr, 0, newArr, 0, index);
+        
+        newArr[index] = numToInsert;
+        
+        System.arraycopy(arr, index, newArr, index + 1, arr.length - index);
+        
+        arr = newArr;
+        return newArr;
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/main/java/com/baeldung/sorted/operations/InsertElementInSortedArray.java:23]:
```suggestion
        return newArr;
```

File: core-java-modules/core-java-arrays-sorting/src/main/java/com/baeldung/sorted/operations/InsertElementInSortedArray.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
+package com.baeldung.sorted.operations;
+
+import java.util.Arrays;
+
+public class InsertElementInSortedArray {
+   
+    public static int[] insertInSortedArr(int[] arr,int numToInsert) {
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/main/java/com/baeldung/sorted/operations/InsertElementInSortedArray.java:7]:
```suggestion
    public static int[] insertInSortedArray(int[] arr, int numToInsert) {
```

File: core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java [hunk 3]
```diff
@@ -0,0 +1,18 @@
+package com.baeldung.sorted.operations;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+public class InsertElementInSortedArrayTest {
+
+    private static final int[] SORTED_ARRAY = new int[] {4,55,85,100,125};
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java:9]:
```suggestion
    private static final int[] SORTED_ARRAY = new int[] {4, 55, 85, 100, 125};
```

File: core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java [hunk 4]
```diff
@@ -0,0 +1,18 @@
+package com.baeldung.sorted.operations;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+public class InsertElementInSortedArrayTest {
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java:7]: Test names must match the convention in 
https://github.com/eugenp/tutorials/blob/ff0f7909e82fb25eba816aaf2db587c0100a4053/custom-pmd/src/main/java/com/baeldung/pmd/UnitTestNamingConventionRule.java#L18

```suggestion
public class InsertElementInSortedArrayUnitTest {
```

File: core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+    private static final int[] SORTED_ARRAY = new int[] {4,55,85,100,125};
+    
+    @Test
+    public void givenSortedArray_whenElementInserted_thenArrayRemainsSorted(){
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java:12]: You have a number of formatting issues - please ensure to apply the Baeldung auto-formatter
```suggestion
    public void givenSortedArray_whenElementInserted_thenArrayRemainsSorted() {
```

File: core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+        int valueToInsert = 90;
+        int[] result = InsertElementInSortedArray.insertInSortedArr(SORTED_ARRAY, valueToInsert);
+        int expectedIndex = 3;
+        assertThat(result[expectedIndex]).isEqualTo(valueToInsert);
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java:16]: Let's assert on the entire array to make sure the other elements were moved to the expected positions too

File: core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java (hunk 4 above)
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayTest.java:7]: Are there edge cases we should be testing? It seems insufficient to provide enough test coverage with a single unit test

**Review by theangrydev — CHANGES_REQUESTED** [review]

File: core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayUnitTest.java [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.jupiter.api.Test;
+
+public class InsertElementInSortedArrayUnitTest {
+
+    @Test
+    void givenSortedArray_whenElementInserted_thenArrayRemainsSorted() {
+        int[] sorted_array = new int[] { 4, 55, 85, 100, 125 };
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayUnitTest.java:11]: In Java, we use camelCase not underscore_care for variable names

File: core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayUnitTest.java [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+        int[] expceted_array = new int[] { 4, 55, 85, 90, 100, 125 };
+        int valueToInsert = 90;
+        int[] result_array = InsertElementInSortedArray.insertInSortedArray(sorted_array, valueToInsert);
```
Comment (theangrydev) [inline core-java-modules/core-java-arrays-sorting/src/test/java/com/baeldung/sorted/operations/InsertElementInSortedArrayUnitTest.java:14]: Let's use a static import for InsertElementInSortedArray.insertInSortedArray

**Review by theangrydev — CHANGES_REQUESTED** [review]

**Review by theangrydev — APPROVED** [review]
